Brokerages/Alpaca/AlpacaBrokerage.Utility.cs

```diff
@@ -0,0 +1,550 @@
+
+            Order order;
+            string tradeEvent = trade.Event.ToUpper();
+            lock (_locker)
+            {
+                order = _orderProvider.GetOrderByBrokerageId(trade.Order.OrderId.ToString());
+            }
+            if (order != null)
+            {
+                if (tradeEvent == "FILL" || tradeEvent == "PARTIAL_FILL")
+                {
+                    order.PriceCurrency = _securityProvider.GetSecurity(order.Symbol).SymbolProperties.QuoteCurrency;
+
+                    var status = Orders.OrderStatus.Filled;
+                    if (trade.Order.FilledQuantity < trade.Order.Quantity) status = Orders.OrderStatus.PartiallyFilled;
+                    OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, 0, "Alpaca Fill Event")
+                    {
+                        Status = status,
+                        FillPrice = trade.Price.Value,
+                        FillQuantity = Convert.ToInt32(trade.Order.Quantity)
+                    });
+                }
+                else if (tradeEvent == "CANCELED")
+                {
+                    OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, 0, "Alpaca Cancel Order Event") { Status = Orders.OrderStatus.Canceled });
+                }
+                else if (tradeEvent == "ORDER_CANCEL_REJECTED")
+                {
+                    Log.Trace($"AlpacaBrokerage.OnTradeUpdate(): Order cancel rejected.");
+                }
+            }
+            else
+            {
+                Log.Error($"AlpacaBrokerage.OnTradeUpdate(): order id not found: {trade.Order.OrderId}");
+            }
+        }
+
+        private void OnNatsClientError(string error)
+        {
+            Log.Error($"NatsClient error: {error}");
+        }
+
+        private void OnSockClientError(Exception exception)
+        {
+            Log.Error(exception, "SockClient error");
+        }
+
+        /// <summary>
+        /// Downloads a list of TradeBars at the requested resolution
+        /// </summary>
+        /// <param name="symbol">The symbol</param>
+        /// <param name="startTimeUtc">The starting time (UTC)</param>
+        /// <param name="endTimeUtc">The ending time (UTC)</param>
+        /// <param name="resolution">The requested resolution</param>
+        /// <param name="requestedTimeZone">The requested timezone for the data</param>
+        /// <returns>The list of bars</returns>
+        private IEnumerable<TradeBar> DownloadTradeBars(Symbol symbol, DateTime startTimeUtc, DateTime endTimeUtc, Resolution resolution, DateTimeZone requestedTimeZone)
+        {
+            // This is due to the polygon API logic.
+            // If start and end date is equal, then the result is null
+            var endTimeUtcForAPI = endTimeUtc.Add(TimeSpan.FromDays(1));
+
+            var period = resolution.ToTimeSpan();
+
+            // No seconds resolution
+            if (period.Seconds < 60)
```

Comment:
Should this use `.TotalSeconds()`?
Probably would be good to add this limit as a comment in the arguments. Does this limit not apply for `DownloadQuoteBars`?

Reply:
Fixed the resolution check for `DownloadTradeBars`, `DownloadQuoteBars` uses a different API method which does not have this limitation.